roles/openstack/nova/tasks/setup_nodes.yml

```diff
@@ -0,0 +1,48 @@
+---
+# This file includes tasks for instances and floating IPs creation.
+
+# Node type can be: controller, compute, ceph, etc.
+- name: Set fact for node type
+  set_fact:
+      node_type: "{{ item }}"
+
+# These tasks done for each type of node according to 'amount' variable.
+- name: Create instances
+  os_server:
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+      auto_ip: "{{ node_type.value.auto_ip | default(auto_ip) }}"
+      config_drive: "{{ node_type.value.config_drive | default(config_drive) }}"
+      flavor: "{{ node_type.value.flavor | default(flavor_id) }}"
+      image: "{{ provisioner.image }}"
+      name: "{{ prefix }}{{ node_type.value.name }}-{{ item }}"
+      nics: "{{ networks.results | map(attribute='id') | list() |map('add_prefix', 'net-id=') | list() }}"
+      key_name: "{{ provisioner.key.name if provisioner.key.name else (prefix + provisioner.key.file | basename) }}"
+      state: present
+      wait: "{{ node_type.value.wait | default(wait) }}"
+  register: nodes
+  with_sequence: "count={{ node_type.value.amount }}"
+  when: provisioner.nodes is defined
+
+- name: Create floating IPs
+  os_floating_ip:
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+      network: "{{ external_network_id }}"
+      internal_network: "{{ prefix}}{{ node_type.value.network.floating_ip_network }}"
+      server: "{{ prefix }}{{ node_type.value.name }}-{{ item }}"
+      state: present
+  register: floating_ips
+  with_sequence: "count={{ node_type.value.amount }}"
+  when: provisioner.nodes is defined
+
+- name: Gather created servers facts
+  os_server_facts:
+      server: "{{ prefix }}{{ node_type.value.name }}*"
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+  register: servers
+- debug: var=servers
+      
+- name: Add created servers to inventory
+  add_host:
+      name:"{{ prefix }}{{ node_type.value.name}}-{{item}}"
+      ansible_ssh_host="{{ servers.ansible_facts.openstack_servers[0].interface_ip }}"
```

Comment:
You are right. But it's a good default

Regards
Yair
On Apr 8, 2016 9:57 AM, "Arie Bregman" <notifications@github.com> wrote:

> In roles/openstack/nova/tasks/setup_nodes.yml
> <https://github.com/rhosqeauto/InfraRed/pull/124#discussion_r58986530>:
>
> > +      state: present
> > +  register: floating_ips
> > +  with_sequence: "count={{ node_type.value.amount }}"
> > +  when: provisioner.nodes is defined
> > +
> > +- name: Gather created servers facts
> > +  os_server_facts:
> > +      server: "{{ prefix }}{{ node_type.value.name }}*"
> > +      cloud: "{{ provisioner.cloud | default(omit) }}"
> > +  register: servers
> > +- debug: var=servers
> > +
> > +- name: Add created servers to inventory
> > +  add_host:
> > +      name:"{{ prefix }}{{ node_type.value.name}}-{{item}}"
> > +      ansible_ssh_host="{{ servers.ansible_facts.openstack_servers[0].interface_ip }}"
>
> why to use fixed value as 'cloud-user' ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/rhosqeauto/InfraRed/pull/124/files/1fa443cf527e8642dcb8b9f726f32842e48c691f..ee7b0d38967d172b1e01560b2538e717aa9d2011#r58986530>
>